components/landing/CodeExamples.tsx

```diff
@@ -22,7 +22,8 @@ export const CodeExamples = () => {
   const isSearching = !!search
   let searchResults: typeof productCodeExamples = []
   if (isSearching) {
-    const matchReg = new RegExp(search, 'i')
+    // The following replace method escapes special characters in regular expression creation.
+    const matchReg = new RegExp(search.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&'), 'i')
```

Comment:
Would it work to just replace anything that isn't `\w`?

Reply:
I'm wondering what the disadvantage of using `includes` instead of `test` below would be, instead of needing to evaluate RegExp. I would also default to an allowlist instead of a denylist.

Reply:
Thanks so much for picking this up @riteshsp2000 ⚡ !  fwiw I tested your PR manually and it resolves the issue that where you would get an error when searching for `*` for example -- but I'll defer to peter and kevin on the discussion above.